internal/controller/utils.go

```diff
@@ -0,0 +1,279 @@
+package controller
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"regexp"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	checkerv1 "github.com/cloudification-io/github-checker-operator/api/v1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var cronJobLimitPointer int32 = 1
+var commonLabelKey string = "cloudification.io/checker"
+var unknownStatus string = "Unknown"
+var httpStatusCodeRegex = regexp.MustCompile(`\b(2\d{2}|3\d{2}|4\d{2}|5\d{2})\b`)
+
+func (r *CheckerReconciler) RenderConfigMap(req *ctrl.Request, checker *checkerv1.Checker) *corev1.ConfigMap {
+	thisConfigMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      checker.ObjectMeta.Name,
+			Namespace: req.Namespace,
+		},
+		Data: map[string]string{
+			"TARGET_URL": checker.Spec.TargetURL,
+		},
+	}
+	return thisConfigMap
+}
+
+func (r *CheckerReconciler) RenderCronJob(req *ctrl.Request, checker *checkerv1.Checker) *batchv1.CronJob {
+	thisCronJob := &batchv1.CronJob{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      checker.ObjectMeta.Name,
+			Namespace: req.Namespace,
+			Labels: map[string]string{
+				commonLabelKey: checker.ObjectMeta.Name,
+			},
+		},
+		Spec: batchv1.CronJobSpec{
+			Schedule:                   "* * * * *",
+			ConcurrencyPolicy:          batchv1.ReplaceConcurrent,
+			SuccessfulJobsHistoryLimit: &cronJobLimitPointer,
+			FailedJobsHistoryLimit:     &cronJobLimitPointer,
+			JobTemplate: batchv1.JobTemplateSpec{
+				Spec: batchv1.JobSpec{
+					Template: corev1.PodTemplateSpec{
+						ObjectMeta: metav1.ObjectMeta{
+							Labels: map[string]string{
+								commonLabelKey: checker.ObjectMeta.Name,
+							},
+						},
+						Spec: corev1.PodSpec{
+							Containers: []corev1.Container{
+								{
+									Name:  "curl",
+									Image: "curlimages/curl:8.10.1",
+									EnvFrom: []corev1.EnvFromSource{
+										{
+											ConfigMapRef: &corev1.ConfigMapEnvSource{
+												LocalObjectReference: corev1.LocalObjectReference{
+													Name: checker.ObjectMeta.Name,
+												},
+											},
+										},
+									},
+									Command: []string{"sh", "-c"},
+									Args: []string{
+										"curl -o /dev/null -s -w \"%{http_code}\" ${TARGET_URL}",
```

Comment:
ok @velp I see 2 things going on, lets start with easier one:

1) `ERROR	Could not get logs from pod	{"error": "container \"curl\" in pod \"github-28804992-lc4k9\" is waiting to start: ContainerCreating"}` looks like I did not properly handle waiting for the pod to reach _Succeded_ phase before fetching logs. I can easily patch that :D 

2) I suspect the issue with curl might be related to DNS or network configuration. I have a general question about your environment - are you running the operator behind some proxy or FW? I need to ask you to provide me some more info about that. Please, run those commands in the pod you have spawned before and paste the output here so I can inspect whats going on:

```
nc -vz github.com 443
nslookup github.com
cat /etc/resolv.conf
```

For me, it looks like there’s a DNS issue or the cluster has been deployed with something like [NodeLocal DNSCache](https://kubernetes.io/docs/tasks/administer-cluster/nodelocaldns/) enabled, which could be causing the problem.

Reply:
1. It cannot handle the situation when the POD in crashloop or when there are 2 containers (one is crashing the second one is creating):
```
$ kgp -n checker
NAME                    READY   STATUS              RESTARTS   AGE
github-28806357-cxct2   0/1     Error               0          35s
github-28806357-gvnlv   0/1     ContainerCreating   0          9s
```
2. results of commands bellow:
```
~ $ nc -vz github.com 443
nc: bad address 'github.com'
~ $ nslookup github.com
Server:		169.254.25.10
Address:	169.254.25.10:53

Non-authoritative answer:

Non-authoritative answer:
Name:	github.com
Address: 140.82.121.4

~ $ cat /etc/resolv.conf
search default.svc.qa.cloudification.io svc.qa.cloudification.io qa.cloudification.io
nameserver 169.254.25.10
options ndots:5
```

but as I said, debugging the DNS problem is out of scope of this task, my question was only about error handling. And even if operator can fetch the logs it will see there `000`:
```
kl -n checker github-28806357-cxct2
log is DEPRECATED and will be removed in a future version. Use logs instead.
000
```

Reply:
ok @velp so: 

1) issue with fetching logs from latest pod before it gets to _Succeded_ phase has been fixed in [1c7051c](https://github.com/cloudification-io/github-checker-operator/pull/2/commits/1c7051c60de811ff5b55f68c290d2cc71a2d8263)

2) whenever the operator fetches `000` from the pod it sets the Checker CRD status to _Unknown_ which in my mind is better way to inspect status of the Checker:

```
λ k get pods
NAME                    READY   STATUS         RESTARTS   AGE
github-28806509-28zmv   0/1     ErrImagePull   0          58s
λ k get checker
NAME     TARGET STATUS
github   Unknown
λ k get pods
NAME                    READY   STATUS      RESTARTS   AGE
github-28806528-qgrrq   0/1     Completed   0          52s
λ k get checker
NAME     TARGET STATUS
github   200
```